python/python/res/simulator/batch_simulator_context.py

```diff
@@ -58,25 +59,32 @@ def results(self):
         when the simulator was created the results will be returned as:
 
 
-          [  {"CMODE" : [1,2,3], "order" : [1,1,3]},
-             {"CMODE" : [1,4,1], "order" : [0,7,8]},
-             {"CMODE" : [6,1,0], "order" : [0,0,8] ]
+          { 0: {"CMODE" : [1,2,3], "order" : [1,1,3]},
+            1: {"CMODE" : [1,4,1], "order" : [0,7,8]},
+            2: {"CMODE" : [6,1,0], "order" : [0,0,8]} }
 
         For a simulation which consist of a total of three "sub-simulations".
 
         """
         if self.running():
             raise RuntimeError("Simulations are still running - need to wait before gettting results")
 
-        res = []
+        res = {}
         nodes = [ EnkfNode(self.res_config.ensemble_config[key]) for key in self.result_keys ]
```

Comment:
This will probably work according to specification; I would still strongly prefer if we checked simulation status with the api, at the very least this will silently return stale data if you are rerunning a simulation, which worked the last time,